### DISCUSSION TITLE
Hiding window decorations in libadwaita

### DISCUSSION THREAD

**Discussion by Pangoraw** [ghostty-org/ghostty#2568] [Feature Requests, Ideas]
When using libadwaita, we manually hide the window header when hiding window decorations. I have been running a patch which only hides this header and does not actually disable window decorations. This keeps the adwaita shadows under the windows and looks nice (subjectively). I would like to discuss having it as a configuration option (`adwaita-hide-window-decorations=all/header`). What do people running without window decorations think?

![image](https://github.com/user-attachments/assets/39fe54f9-c8c4-4ea7-bdeb-05af51797929)

**Comment by mitchellh** [comment]
From my perspective: I'm fine with this, but would love to hear from folks that use this config since they seem to be really opinionated to make it work with their window managers and that's just not an area I have a lot of experience with.

For `adwaita-hide-window-decorations`, we can consider making it a `packed struct` of bools which will allow a config syntax like this (already works): `header,no-header` etc. and `false` disables _all_.

**Reply by yotamN** [reply]
As a GNOME user I think that would be really awesome. I use the no window decoration option because I want maximum screen real estate but it does look a bit ugly in its current form.

**Comment by ghost** [comment]
It looks amazing, but I imagine one would lose ability to drag the window around. If you somehow come up with a way to drag the window that is not cumbersome, I'd love to use the terminal this way.

**Reply by yotamN** [reply]
GNOME (and other DEs I believe) support a keybinding to drag around windows without a titlebar. `ctrl+left button` would let you drag the window and `ctrl+middle button` would let you resize it.

**Reply by jcollie** [reply]
You can also use Gnome to move/resize Ghostty's windows strictly with the keyboard. I verified that the following work:

`alt+f7` lets you move a window
`alt+f8` lets you resize a window
`alt+f10` lets you toggle maximize
`alt+space` brings up a menu with options for moving/resizing a window

https://askubuntu.com/questions/18243/moving-windows-from-the-keyboard

**Reply by sionescu** [reply]
You can drag a window using Alt and left mouse click anywhere inside a window.

**Reply by ghost** [reply]
Those all are suboptimal approaches, as dragging and resizing windows is a cursor bound action and proposed workarounds involve keyboard.

**Reply by ghost** [reply]
What could possibly work is if Ghostty team @mitchellh implemented something like middle click drag.

**Comment by IrishBruse** [comment]
+1 this doing the same its nice being able to resize with the mouse
also if doing it proper i noticed the resize popup is touching the edge which looks a little off some padding for the popup would look good
![image](https://github.com/user-attachments/assets/fe2e056e-c24c-4516-bd91-f330f735095b)

**Comment by alosarjos** [comment]
This looks really great. Is the headerbar also hidden on fullscreen mode?

**Reply by jcollie** [reply]
Now that #5008 is merged the titlebar will be hidden when the window goes into fullscreen mode.

**Reply by alosarjos** [reply]
Oh great!!!

Still I really like the style on windowed mode presented here.

**Reply by jcollie** [reply]
Hiding _just_ the titlebar is already possible by setting `gtk-titlebar=false`. The window will keep the other decorations, like rounded corners and the drop shadow.
